test/CMakeLists.txt

```diff
@@ -5,6 +5,28 @@
 # The complete license agreement can be obtained at:
 # http://arrayfire.com/licenses/BSD-3-Clause
 
+
+set(BUILD_DIR "${ArrayFire_BINARY_DIR}")
+add_definitions("-DBUILD_DIR=\"${BUILD_DIR}\"")
+
+set(CPU_LIB_NAME "${CMAKE_SHARED_LIBRARY_PREFIX}afcpu${CMAKE_SHARED_LIBRARY_SUFFIX}")
+set(CUDA_LIB_NAME "${CMAKE_SHARED_LIBRARY_PREFIX}afcuda${CMAKE_SHARED_LIBRARY_SUFFIX}")
+set(OPENCL_LIB_NAME "${CMAKE_SHARED_LIBRARY_PREFIX}afopencl${CMAKE_SHARED_LIBRARY_SUFFIX}")
+
+if(CMAKE_RUNTIME_OUTPUT_DIRECTORY)
+  set(CPU_LIB_PATH "${CMAKE_RUNTIME_OUTPUT_DIRECTORY}/${CPU_LIB_NAME}")
+  set(CUDA_LIB_PATH "${CMAKE_RUNTIME_OUTPUT_DIRECTORY}/${CUDA_LIB_NAME}")
+  set(OPENCL_LIB_PATH "${CMAKE_RUNTIME_OUTPUT_DIRECTORY}/${OPENCL_LIB_NAME}")
+else()
+  set(CPU_LIB_PATH "${ArrayFire_BINARY_DIR}/${CPU_BACKEND_SOURCE_DIR}/${CPU_LIB_NAME}")
+  set(CUDA_LIB_PATH "${ArrayFire_BINARY_DIR}/${CUDA_BACKEND_SOURCE_DIR}/${CUDA_LIB_NAME}")
+  set(OPENCL_LIB_PATH "${ArrayFire_BINARY_DIR}/${OPENCL_BACKEND_SOURCE_DIR}/${OPENCL_LIB_NAME}")
+endif()
```

Comment:
Nit:

```cmake
set(LIB_PATH "${ArrayFire_BINARY_DIR}/${CPU_BACKEND_SOURCE_DIR}")
if(CMAKE_RUNTIME_OUTPUT_DIRECTORY)
  set(LIB_PATH "${CMAKE_RUNTIME_OUTPUT_DIRECTORY}")
endif()
set(CPU_LIB_PATH "${LIB_PATH}/${CPU_LIB_NAME}")
set(CUDA_LIB_PATH "${LIB_PATH}/${CUDA_LIB_NAME}")
set(OPENCL_LIB_PATH "${LIB_PATH}/${OPENCL_LIB_NAME}")
```
is couple of lines shorter than
```cmake
if(CMAKE_RUNTIME_OUTPUT_DIRECTORY)
  set(CPU_LIB_PATH "${CMAKE_RUNTIME_OUTPUT_DIRECTORY}/${CPU_LIB_NAME}")
  set(CUDA_LIB_PATH "${CMAKE_RUNTIME_OUTPUT_DIRECTORY}/${CUDA_LIB_NAME}")
  set(OPENCL_LIB_PATH "${CMAKE_RUNTIME_OUTPUT_DIRECTORY}/${OPENCL_LIB_NAME}")
else()
  set(CPU_LIB_PATH "${ArrayFire_BINARY_DIR}/${CPU_BACKEND_SOURCE_DIR}/${CPU_LIB_NAME}")
  set(CUDA_LIB_PATH "${ArrayFire_BINARY_DIR}/${CUDA_BACKEND_SOURCE_DIR}/${CUDA_LIB_NAME}")
  set(OPENCL_LIB_PATH "${ArrayFire_BINARY_DIR}/${OPENCL_BACKEND_SOURCE_DIR}/${OPENCL_LIB_NAME}")
endif()
```

Reply:
Yes that's shorter indeed. However, we can't really use the same `LIB_PATH` variable for all the backends at the end - that may be true for Win32 builds where we set `CMAKE_RUNTIME_OUTPUT_DIRECTORY`, but for everything else, it is empty and therefore CMake puts the output libraries in the default location: `build/src/backend/<backend name>`. This is why I have something like `${ArrayFire_BINARY_DIR}/${CPU_BACKEND_SOURCE_DIR}` before each library name in the else clause.